### ISSUE TITLE
tmux layer: Undefined - statusline_inactive_separator

### ISSUE THREAD

**Issue by msphinct** [wsdjeg/SpaceVim#3343]
## Expected behavior, english is required
```

Distributor ID: Ubuntu
Description:    Ubuntu 19.10
Release:        19.10
Codename:       eoan
Linux  5.3.0-40-lowlatency #32- Ubuntu SMP PREEMPT 
Spacevim 1.3.0 & Spacevim 1.4.0.dev
NVIM v0.5.0  (still broken on 0.4.3 and earlier)
```

I might be doing something wrong, but I think there is a missing default definition in the **tmux layer.** This isn't a huge deal, but it should be an very easy fix:

With the **tmux layer** enabled, an error is thrown at startup. I believe it's because the _statusline_inactive_separator_ is ultimitly Undefined. SPUpdate fails.  I've included a video at the bottom.  


Temporary fix: Define _statusline_inactive_separator_ in SpaceVim.d/init.toml

## The reproduce ways from Vim starting (Required!)

```
1. rm -f .cache/vimfiles
2. cat  .SpaceVim/SpaceVim.d/init.toml


[options]                                                                                                                                                     
                                                                                                                                                     
checkinstall = 1                                                                                                                                          
vimcompatible = 1                                                                                                                                         
                                                                                                                                                       [[layers]]                                                                                                                                                    name = "tmux"        
```

3  Launch vim

```
> vim 

## Debug info
Error detected while processing /home/pete/.SpaceVim/autoload/SpaceVim/layers/tmux.vim:
line  121:
E121: Undefined variable: g:spacevim_statusline_inactive_separator
E15: Invalid expression: g:spacevim_statusline_inactive_separator
Error detected while processing function SpaceVim#custom#load[16]..SpaceVim#custom#apply:
line   25:
E171: Missing :endif
Error detected while processing function SpaceVim#custom#load:
line   16:
E171: Missing :endif


```
I cannot display plugins at this point, because nothing loads. After the temporary fix, below, the plugins will load and I display them.

Temporary Fix: define  _statusline_inactive_separator_
```
[options]
statusline_inactive_separator='arrow'
```

**AFTER** restarting, everything loads, and now plugins display too

SpaceVim layers:

```
- VersionControl:       not loaded          https://spacevim.org/layers/VersionControl/
+ autocomplete:             loaded          https://spacevim.org/layers/autocomplete/
- chat:                 not loaded          https://spacevim.org/layers/chat/
+ checkers:                 loaded          https://spacevim.org/layers/checkers/
- chinese:              not loaded          https://spacevim.org/layers/chinese/
- colorscheme:          not loaded          https://spacevim.org/layers/colorscheme/
+ core#banner:              loaded          https://spacevim.org/layers/core/banner/
+ core#statusline:          loaded          https://spacevim.org/layers/core/statusline/
+ core#tabline:             loaded          https://spacevim.org/layers/core/tabline/
+ core:                     loaded          https://spacevim.org/layers/core/
- cscope:               not loaded          https://spacevim.org/layers/cscope/
- ctrlp:                not loaded          https://spacevim.org/layers/ctrlp/
- ctrlspace:            not loaded          https://spacevim.org/layers/ctrlspace/
- debug:                not loaded          https://spacevim.org/layers/debug/
- denite:               not loaded          https://spacevim.org/layers/denite/
+ edit:                     loaded          https://spacevim.org/layers/edit/
- exprfold:             not loaded          no exists
- floobits:             not loaded          https://spacevim.org/layers/floobits/
+ format:                   loaded          https://spacevim.org/layers/format/
- fzf:                  not loaded          https://spacevim.org/layers/fzf/
- games:                not loaded          no exists
- git:                  not loaded          https://spacevim.org/layers/git/
- github:               not loaded          https://spacevim.org/layers/github/
- gtags:                not loaded          https://spacevim.org/layers/gtags/
- incsearch:            not loaded          no exists
- indentmove:           not loaded          no exists
- japanese:             not loaded          https://spacevim.org/layers/japanese/
- lang#WebAssembly:     not loaded          https://spacevim.org/layers/lang/WebAssembly/
- lang#agda:            not loaded          https://spacevim.org/layers/lang/agda/
- lang#asciidoc:        not loaded          https://spacevim.org/layers/lang/asciidoc/
- lang#assembly:        not loaded          https://spacevim.org/layers/lang/assembly/
- lang#autohotkey:      not loaded          https://spacevim.org/layers/lang/autohotkey/
- lang#batch:           not loaded          https://spacevim.org/layers/lang/batch/
- lang#c:               not loaded          https://spacevim.org/layers/lang/c/
- lang#chapel:          not loaded          https://spacevim.org/layers/lang/chapel/
- lang#clojure:         not loaded          https://spacevim.org/layers/lang/clojure/
- lang#coffeescript:    not loaded          https://spacevim.org/layers/lang/coffeescript/
- lang#crystal:         not loaded          https://spacevim.org/layers/lang/crystal/
- lang#csharp:          not loaded          https://spacevim.org/layers/lang/csharp/
- lang#d:               not loaded          https://spacevim.org/layers/lang/d/
- lang#dart:            not loaded          https://spacevim.org/layers/lang/dart/
- lang#dockerfile:      not loaded          https://spacevim.org/layers/lang/dockerfile/
- lang#elixir:          not loaded          https://spacevim.org/layers/lang/elixir/
- lang#elm:             not loaded          https://spacevim.org/layers/lang/elm/
- lang#erlang:          not loaded          https://spacevim.org/layers/lang/erlang/
- lang#extra:           not loaded          https://spacevim.org/layers/lang/extra/
- lang#foxpro:          not loaded          https://spacevim.org/layers/lang/foxpro/
- lang#fsharp:          not loaded          https://spacevim.org/layers/lang/fsharp/
- lang#go:              not loaded          https://spacevim.org/layers/lang/go/
- lang#goby:            not loaded          https://spacevim.org/layers/lang/goby/
- lang#gosu:            not loaded          https://spacevim.org/layers/lang/gosu/
- lang#graphql:         not loaded          https://spacevim.org/layers/lang/graphql/
- lang#groovy:          not loaded          https://spacevim.org/layers/lang/groovy/
- lang#hack:            not loaded          https://spacevim.org/layers/lang/hack/
- lang#haskell:         not loaded          https://spacevim.org/layers/lang/haskell/
- lang#html:            not loaded          https://spacevim.org/layers/lang/html/
- lang#hy:              not loaded          https://spacevim.org/layers/lang/hy/
- lang#idris:           not loaded          https://spacevim.org/layers/lang/idris/
- lang#io:              not loaded          https://spacevim.org/layers/lang/io/
- lang#ipynb:           not loaded          https://spacevim.org/layers/lang/ipynb/
- lang#j:               not loaded          https://spacevim.org/layers/lang/j/
- lang#java:            not loaded          https://spacevim.org/layers/lang/java/
- lang#javascript:      not loaded          https://spacevim.org/layers/lang/javascript/
- lang#json:            not loaded          no exists
- lang#julia:           not loaded          https://spacevim.org/layers/lang/julia/
- lang#kotlin:          not loaded          https://spacevim.org/layers/lang/kotlin/
- lang#latex:           not loaded          https://spacevim.org/layers/lang/latex/
- lang#lisp:            not loaded          https://spacevim.org/layers/lang/lisp/
- lang#livescript:      not loaded          https://spacevim.org/layers/lang/livescript/
- lang#lua:             not loaded          https://spacevim.org/layers/lang/lua/
- lang#markdown:        not loaded          https://spacevim.org/layers/lang/markdown/
- lang#matlab:          not loaded          https://spacevim.org/layers/lang/matlab/
- lang#nim:             not loaded          https://spacevim.org/layers/lang/nim/
- lang#nix:             not loaded          https://spacevim.org/layers/lang/nix/
- lang#ocaml:           not loaded          https://spacevim.org/layers/lang/ocaml/
- lang#pact:            not loaded          https://spacevim.org/layers/lang/pact/
- lang#pascal:          not loaded          no exists
- lang#perl:            not loaded          https://spacevim.org/layers/lang/perl/
- lang#php:             not loaded          https://spacevim.org/layers/lang/php/
- lang#plantuml:        not loaded          https://spacevim.org/layers/lang/plantuml/
- lang#pony:            not loaded          https://spacevim.org/layers/lang/pony/
- lang#powershell:      not loaded          https://spacevim.org/layers/lang/powershell/
- lang#processing:      not loaded          https://spacevim.org/layers/lang/processing/
- lang#prolog:          not loaded          https://spacevim.org/layers/lang/prolog/
- lang#puppet:          not loaded          https://spacevim.org/layers/lang/puppet/
- lang#purescript:      not loaded          https://spacevim.org/layers/lang/purescript/
- lang#python:          not loaded          https://spacevim.org/layers/lang/python/
- lang#r:               not loaded          https://spacevim.org/layers/lang/r/
- lang#racket:          not loaded          https://spacevim.org/layers/lang/racket/
- lang#red:             not loaded          https://spacevim.org/layers/lang/red/
- lang#ruby:            not loaded          https://spacevim.org/layers/lang/ruby/
- lang#rust:            not loaded          https://spacevim.org/layers/lang/rust/
- lang#scala:           not loaded          https://spacevim.org/layers/lang/scala/
- lang#scheme:          not loaded          https://spacevim.org/layers/lang/scheme/
- lang#sh:              not loaded          https://spacevim.org/layers/lang/sh/
- lang#slim:            not loaded          no exists
- lang#solidity:        not loaded          no exists
- lang#sql:             not loaded          no exists
- lang#swift:           not loaded          https://spacevim.org/layers/lang/swift/
- lang#swig:            not loaded          no exists
- lang#tcl:             not loaded          https://spacevim.org/layers/lang/tcl/
- lang#toml:            not loaded          no exists
- lang#typescript:      not loaded          https://spacevim.org/layers/lang/typescript/
- lang#v:               not loaded          https://spacevim.org/layers/lang/v/
- lang#vim:             not loaded          https://spacevim.org/layers/lang/vim/
- lang#vue:             not loaded          https://spacevim.org/layers/lang/vue/
- lang#wolfram:         not loaded          https://spacevim.org/layers/lang/wolfram/
- lang#xml:             not loaded          no exists
- leaderf:              not loaded          https://spacevim.org/layers/leaderf/
- lsp:                  not loaded          no exists
- mail:                 not loaded          no exists
- operator:             not loaded          no exists
- org:                  not loaded          no exists
- shell:                not loaded          https://spacevim.org/layers/shell/
- sudo:                 not loaded          https://spacevim.org/layers/sudo/
- test:                 not loaded          https://spacevim.org/layers/test/
+ tmux:                     loaded          https://spacevim.org/layers/tmux/
- tools#dash:           not loaded          https://spacevim.org/layers/tools/dash/
- tools#mpv:            not loaded          no exists
- tools#screensaver:    not loaded          no exists
- tools#zeal:           not loaded          https://spacevim.org/layers/tools/zeal/
- tools:                not loaded          https://spacevim.org/layers/tools/
+ ui:                       loaded          https://spacevim.org/layers/ui/
- unite:                not loaded          https://spacevim.org/layers/unite/
- vim:                  not loaded          no exists

```

## Screenshots

## Debug info

Here's a clear example.

[Video demonstration](https://asciinema.org/a/L38jNgPnhtheyBmvfFjh3oQVl)

Thanks for all of you work!

**Comment by ghost** [comment]
#3342 should resolve this issue.

**Comment by wsdjeg** [comment]
@hermitmaster thank you, I have merged your PR. 

@msphinct can you have a try?

**Comment by msphinct** [comment]
Hey guys!  I just pulled the latest, but when I ran it a fresh install with
just the bare init.toml, I still get:
rror detected while processing function
SpaceVim#custom#load[16]..SpaceVim#custom#apply[25]..SpaceVim#layers#load[9]..SpaceVim#layers#tmux#set_var
iable:
line    6:
E121: Undefined variable: g:spacevim_statusline_inactive_separator
E116: Invalid arguments for function get
E15: Invalid expression: get(a:var, 'tmuxline_separators_alt',
g:spacevim_statusline_inactive_separator)
Error detected while processing function
SpaceVim#custom#load[16]..SpaceVim#custom#apply:
line   25:
E171: Missing :endif
Error detected while processing function SpaceVim#custom#load:
line   16:
E171: Missing :endif


Here's my revision:
--

commit 76afe2650bd15c057513e7e885a82537bf731d8a (HEAD -> master,
origin/master, origin/HEAD)
Author: Dennis Rausch <hermitmaster@icloud.com>
Date:   Mon Feb 10 22:39:45 2020 -0600

    fix: Make tmux layer use statusline_iseparator and added separators.
(#3342)
---
Thoughts?

Thanks!

Pete

**Comment by msphinct** [comment]
Just in case, here's my init.toml:

[options]

    checkinstall = 1
     vimcompatible = 1

 [[layers]]
 name = "tmux"

**Comment by wsdjeg** [comment]
Please update spacevim, this has been fixed in #3342

edit:

sorry, I have fixed it in https://github.com/SpaceVim/SpaceVim/commit/113eba351d5b3a07039e7d657ae1ca994a54f2ad

**Comment by msphinct** [comment]
Awesome, works great, thanks again!
